### ISSUE TITLE
Schema handling in dialects.sqlite.py

### ISSUE THREAD

**Issue by rontroad** [geoalchemy/geoalchemy2#423]
dialiects.sqlite.py before_drop and after_create are setting the schema name as argument for geometry column operation.

It seems strange to me as from my understanding (and runtime exceptions) sqlite/spatialite doesn't handle schema.

It should at least take into account sqlalchemy schema translate map

code in before_drop and after_create :
`args = [table.schema] if table.schema else []`
may be removed or changed into
`table_schema = bind.schema_for_object(table)`
`args = [table_schema] if table_schema else []`

My use case is trying to mix postgis and spatialite as datasource mainly for testing purpose.

I can help on that matter if you confirm the issue.

**Comment by adrien-berchet** [comment]
Hi @rontroad , thank you for this report!
It seems you're right, it makes not sense to add this schema for SQLite. Nevertheless, if a table is defined with a schema, it will fail anyway with SQLite dialect, isn't it?

For example, this fails even after fixing the `before_drop` and `after_create` functions:
```python
        # Define the table
        t = Table(
            "a_table",
            MetaData(),
            Column("id", Integer, primary_key=True),
            Column("geom", Geometry()),
            schema="gis"
        )

        # Create the table
        t.create(bind=conn)
```
with the following error:
```Python traceback
self = <sqlalchemy.dialects.sqlite.pysqlite.SQLiteDialect_pysqlite object at 0x7fa426c44c10>, cursor = <sqlite3.Cursor object at 0x7fa4271dfb20>
statement = '\nCREATE TABLE gis.no_geom_type (\n\tid INTEGER NOT NULL, \n\tgeom GEOMETRY, \n\tPRIMARY KEY (id)\n)\n\n', parameters = ()
context = <sqlalchemy.dialects.sqlite.base.SQLiteExecutionContext object at 0x7fa426ccc5b0>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unknown database gis
E       [SQL: 
E       CREATE TABLE gis.no_geom_type (
E       	id INTEGER NOT NULL, 
E       	geom GEOMETRY, 
E       	PRIMARY KEY (id)
E       )
E       
E       ]
E       (Background on this error at: https://sqlalche.me/e/14/e3q8)

.tox/py38-sqla14/lib/python3.8/site-packages/sqlalchemy/engine/default.py:736: OperationalError
```

**Comment by rontroad** [comment]
Hi,

sqlalchemy has a schema translation option
https://docs.sqlalchemy.org/en/20/core/connections.html#translation-of-schema-names
that can be set on the engine creation or the connection. The connection
class has the schema_for_object(table) method to take this map into account.

The example should work with the extra option in create_engine( ...,
execution_options={"schema_translate_map": {"gis": None}})

The final purpose would be to use the same ORM model with postgis and
sqlite.

For information, I plan to try using the sqlite ATTACH statement to
actually make a sql request using schema names working with sqlite. (but
I'm stuck now because my model geometry has a M dimension and it seems that
geoalchemy retrieves only XY from postgis and so it fails during the
insertion, I'll have to check and dig a little bit more)

Le ven. 3 févr. 2023 à 18:48, Adrien Berchet ***@***.***> a
écrit :

**Comment by adrien-berchet** [comment]
Ah nice, I didn't know the `schema_translate_map` option, this is very convenient.
Anyway, I removed the schema in #424 , as the SpatiaLite functions do not accept the schema anyway. Can you test your use case with the master branch? If it works properly I will release a new version.

**Comment by rontroad** [comment]
Works for me

**Comment by adrien-berchet** [comment]
Ok, thanks @rontroad !

**Comment by adrien-berchet** [comment]
I just released new 0.13.2 version which should work for you @rontroad
